src/Tools.js

```diff
@@ -12,19 +12,20 @@ class Tools {
 	 * within a bidi isolation block.
 	 *
 	 * @param {jQuery|string} $content Content to isolate
-	 * @param {boolean} returnRawHtml Force the return value to
+	 * @param {boolean} [returnRawHtml] Force the return value to
 	 *  be raw html. If set to false, will return the encompassed
 	 *  jQuery object.
 	 * @return {jQuery|string} BiDi isolated jQuery object or HTML
 	 */
-	static bidiIsolate( $content, returnRawHtml ) {
+	static bidiIsolate( $content, returnRawHtml = false ) {
 		const $result = $( '<bdi>' );
+
 		if ( typeof $content === 'string' ) {
-			$result.text( $content );
-		} else {
-			$result.append( $content );
+			$content = $( $.parseHTML( $content ) );
```

Comment:
So, I think we have different perspectives on the role of the function and the angle of where the future-proofing comes from.

Just as a general comment, `$.parseHTML()` is not a safety check, it just parses the string. We are only shifting the responsibility of verifying safety from the caller to this method.

With your approach (and please correct me if I've understood it wrong) you set things up where the method outputs the "correct" value all the time, no matter what the user of the method gives it. That means that we prioritize the ease of the user of the method (and make all uses of it simpler), but make the code more vague about how it protects data that may (perhaps even in the future usage of this method) not be as we expect.

With my approach, we intentionally put the responsibility of what is sent and how to handle it, specifically, on the user of the method. This means that I prioritize the generalization of the method at the "expense" of the user of the method, but it also means that the place where we decide how to handle a given data is the place where the data is in its context, which makes it a lot easier for us to note edge cases and to "convince ourselves" (and others who read the code in x years) which data provides what kind of functionality.

It also means that it exposes edge cases much quicker, which forces the user of the function to decide whether these uses deserve a different way of handling its data.

For example, let's say we collect a certain data element from some API request. Even if we assume that all API requests are safe (which is not necessarily always true, but true enough for now) we will send that API request as string to the wrapping method.

If we go with my approach, we will immediately see that there's something unique about that data that we may not have immediately expected or anticipated because all HTML would appear, immediately, unparsed. We'd look at it and know that we need to parse the HTML, that the data piece *may contain* html (or wikitext) and have a chance to decide how secure or insecure it is.

With your approach, the string will work out of the box -- which is convenient -- but it would also mean we risk missing those steps of being certain of every piece of data we're handling.

I don't think either one of us is wrong or right. I think it's a matter of priorities of where we see the different pieces of the code operates. For me personally, I prefer strategies that, while may look a tad bit more verbose at times, are also making the entire code immediately clear about what kind of data it handles, even if we look back at this code years and years from now, or even if we iterate on code and keep adding different features where snippets that are given to us are from various sources.

I'm okay going with your method (I prefer mine but I won't block merging over this preference ;) ) but I do think that if we *do* go with yours, we should compensate by having a comment written on the "user" side (the code user, where we call this method) that specifies whether we get HTML or plaintext, for posterity.

Reply:
Another piece that occurred to me now is that with my method, we can actually give more control to the caller; so, if you know that you have a string that might *seem* like html but you want it to not render, you intentionally send it in as a string. With the change you're proposing, there's basically no way for you to purposefully do that.

An example would be a username that's something like `&nbsp;`. Those may well be legal characters for a user (I'm not sure what we're using in Wikipedias, but according to [$wgInvalidUsernameCharacters](https://www.mediawiki.org/wiki/Manual:$wgInvalidUsernameCharacters) the default illegal characters are just `@` and `:`. So, if you have that user, you actually want it to appear as `&nbps;` and not a space. But if we take all strings and HTMLify them, we will never be able to effectively do that unless we force them through htmlentities or something similar. I think that might be the case with our username policy, though I'm not sure (marked it below for your consideration)  

Again, I think there's no real right or wrong here. I am okay changing things if you think it works better, and if we do encouter the above cases, we can edge case them specifically, too.

Reply:
You might be overthinking this! There is no "my way", I just needed it not escape the HTML. You told me to wrap it in a jQuery object, so I did. The function you wrote allowed you to pass in a string, so I kept that. However you want to do is fine with me. What is your suggestion?

> An example would be a username that's something like `&nbsp;`. ... according to $wgInvalidUsernameCharacters the default illegal characters are just `@` and `:`

Usernames also are also bound by [$wgLegalTitleChars](https://www.mediawiki.org/wiki/Manual:$wgLegalTitleChars), since they can have a user/user talk pages. This prohibits `#<>[]|{}`. HTML entities (such as `&nbsp;`) are not allowed, either.

Reply:
Oops, I think I wasn't clear. I meant that the **caller** should wrap the content it sends to this method with a jquery method, not that the method itself should do it (For the reasons I put up above).

And I hope this didn't come off too abruptly -- I didn't mean to suggest we're batting heads; I meant more like "this is the approach I'm taking, while this is where you're coming from". Both are totally valid, but since there's no clear "this is wrong or right" I just wanted to explain why I prefer the caller to do the work.

That said, this is perfectly valid. Thank you for putting up with my attempts at explaining the jumbled thoughts in my head ;) 

I need to retest this and then it can be merged.